repo-issues/lib/data-source.js

```diff
@@ -0,0 +1,97 @@
+const http = require('request-promise-native')
+const moment = require('moment')
+const octokit = require('@octokit/rest')()
+
+const show_detaiils_threshold = 5
+const organisation = 'novoda'
+const languageIconMap = {
+    "html": "html5",
+    "groovy": "gradle",
+    "css": "css3",
+    "objective-c" : "c"
+}
+
+var showIssues = false
+var showPRs = true
+
+const generateViewState = (configuration) => {
+    showIssues = configuration.showIssues
+    showPRs = configuration.showPRs
+
+    octokit.authenticate({type: 'oauth', token: `${configuration.token}`})
+    
+    return octokit.repos.getForOrg({ org: organisation, type: 'all', per_page: 9999 })
+        .then(repos => Promise.all(repos.data.map(queryRepoIssues)))
+        .then(prs => prs.filter(each => each.prs.length > 0 || each.issues.length > 0))
+        .then(toViewState)
+}
+
+function queryRepoIssues(repo) {
+    return octokit.issues.getForRepo({ owner: organisation, repo: repo.name })
+        .then(result => toRepoIssues(result, repo))
+}
+
+function toRepoIssues(result, repo) {
+    const allitems = { prs: [], issues: [] }
+
+    result.data
+        .filter(each =>
```

Comment:
I am confused by this api ^^
So `octokit.issues.getForRepo` returns `Issues` together with `PullRequests`? Or is `each.pull_request` maybe just the PR  where the current `Issue` is linked? That would mean we just consider PR's that are linked in issues 🤔

Reply:
Issues together with PRs. In Github API everything is considered "issues" (regular issues and pull requests) - That's why I need to do some filtering logic to divide them in two separate arrays